packages/app/src/app/components/WorkspaceSetup/StepHeader.tsx

```diff
@@ -15,48 +17,56 @@ export const StepHeader = ({
   title,
   onDismiss,
   onPrevStep,
-}) => (
-  <Stack direction="vertical" gap={8}>
-    {currentStep === 0 ? (
-      <Button autoWidth variant="secondary" onClick={onDismiss}>
-        <Stack gap={2}>
-          <Icon
-            name="arrowDown"
-            css={{ transform: 'rotate(90deg)' }}
-            size={16}
-          />
-          <Text>Cancel</Text>
-        </Stack>
-      </Button>
-    ) : (
-      <Button autoWidth variant="secondary" onClick={onPrevStep}>
-        <Stack gap={2}>
-          <Icon
-            name="arrowDown"
-            css={{ transform: 'rotate(90deg)' }}
-            size={16}
-          />
-          <Text>Back</Text>
-        </Stack>
-      </Button>
-    )}
-    <Stack direction="vertical" gap={1}>
-      {numberOfSteps > 1 && (
-        <Text size={4} weight="medium" color="#fff">
-          STEP {currentStep + 1} OF {numberOfSteps}
-        </Text>
+  workspaceId,
+}: StepHeaderProps) => {
+  const { dashboard } = useAppState();
+  const workspaceName = dashboard.teams.find(t => t.id === workspaceId)?.name;
+
+  return (
+    <Stack direction="vertical" gap={8}>
+      {currentStep === 0 ? (
+        <Button autoWidth variant="secondary" onClick={onDismiss}>
+          <Stack gap={2}>
+            <Icon
+              name="arrowDown"
+              css={{ transform: 'rotate(90deg)' }}
+              size={16}
+            />
+            <Text>Cancel</Text>
+          </Stack>
+        </Button>
+      ) : (
+        <Button autoWidth variant="secondary" onClick={onPrevStep}>
+          <Stack gap={2}>
+            <Icon
+              name="arrowDown"
+              css={{ transform: 'rotate(90deg)' }}
+              size={16}
+            />
+            <Text>Back</Text>
+          </Stack>
+        </Button>
       )}
+      <Stack direction="vertical" gap={1}>
+        {numberOfSteps > 1 && (
+          <Text size={4} weight="medium" color="#fff">
+            STEP {currentStep + 1} OF {numberOfSteps}
+          </Text>
+        )}
+
+        <Text
+          margin={0}
+          as="h1"
+          color="#fff"
+          weight="medium"
+          fontFamily="everett"
+          size={32}
+        >
+          {title}
+        </Text>
 
-      <Text
-        margin={0}
-        as="h1"
-        color="#fff"
-        weight="medium"
-        fontFamily="everett"
-        size={32}
-      >
-        {title}
-      </Text>
+        {workspaceName && <Text>Upgrade {workspaceName}&apos;s workspace</Text>}
```

Comment:
The same component is reused when you create a new workspace, so `Upgrade` will not work here. I think it's ok to use just the workspace name

Reply:
This caption only shows up after the create and select step, but the implementation didn't make it clear